pennylane/pulse/parametrized_evolution.py

```diff
@@ -61,9 +62,22 @@ class ParametrizedEvolution(Operation):
             the parameters of a scalar-valued function of the Hamiltonian being evolved.
         t (Union[float, List[float]]): If a float, it corresponds to the duration of the evolution.
             If a list of floats, the ODE solver will use all the provided time values, and
-            perform intermediate steps if necessary. It is recommended to just provide a start and end time.
-            Note that such absolute times only have meaning within an instance of
+            perform intermediate steps if necessary. It is recommended to just provide a start
+            and end time unless matrices of the time evolution at intermediate times needs
+            to be computed. Note that such absolute times only have meaning within an instance of
             ``ParametrizedEvolution`` and will not affect other gates.
+        return_intermediate (bool): Whether or not the ``matrix`` method returns all intermediate
+            solutions of the time evolution ODE at the times provided in ``t``. If ``False``
+            (the default), only the matrix for the full time evolution is returned. If ``True``,
+            all solutions after :math:`U(t_1, t_1)=1` are returned; when used in a circuit, this
+            results in ``ParametrizedEvolution`` being a broadcasted operation.
+        complementary (bool): Whether or not to compute the complementary time evolution when using
+            ``return_intermediate=True`` (ignored otherwise).
+            If ``False`` (the default), the usual solutions to the Schrodinger equation
+            :math:`\{U(t_1, t_1), U(t_1, \tau_1),\dots, U(t_1, \tau_P), U(t_1, t_2)\}` are computed,
+            where :math:`\tau_i` are the additional times provided in ``t``.
```

Comment:
```suggestion
            where :math:`\tau_i` are the additional times provided in ``t = [t1, tau1, tau2, .., tauN, t2]``.
```

Something like this. Though I personally find the taus here a bit confusing, I would maybe suggest to just make it clear that `t=[t0, t1, t2, t3, .. , tN]` and use the appropriate times in the docstring

Reply:
Good suggestion! :+1: 
Do we want 0-based or 1-based indexing of the times? I think the docs mostly use `t1` and `t2`, right? For a longer list, however, starting with 1 feels a bit funny in a Python docstring :see_no_evil:

Reply:
I would definitely say from 0 for that reason 😄

Reply:
Okay, this might lead to changed docs across the pulse module, I believe? I'll check and add those changes to this PR.